orbit_particles.c

```diff
@@ -714,6 +721,11 @@ void do_particle_kernel(Config_t* cfg_ptr, int particle_id){
     konestep(cfg_ptr, particle_id);
     kupdate(cfg_ptr, particle_id);
 
+    /* XXX should this go before onestep and update?... */
```

Comment:
I think this is the correct order: first push particles, then compute mode evolution based on interaction between modes and particles -> update mode amplitude for next step.